### PR TITLE
[fix] set unrestricted not working correctly

### PR DESCRIPTION
**Pull request by kopancek**
## Description

Two problems fixed here:
1. When using unified perms table, the SQL call was removing all rows where repo_id = %d. This is not correct, as we might have other rows in the DB with the same repo_id
2. When sending more items than parameter limit, the query would fail. Fixed by doing requests in chunks.

## Test plan

unit tests added/modified

### CODE REVIEW

File: enterprise/internal/database/perms_store.go [hunk 1]
```diff
@@ -871,9 +871,21 @@ DO UPDATE SET
    unrestricted = %s;
 `
 
-	q := sqlf.Sprintf(format, pq.Array(ids), unrestricted, unrestricted)
+	size := 65535 / 2 // 65535 is the max number of parameters in a query, 2 is the number of parameters in each row
```
Comment (vdavid) [inline enterprise/internal/database/perms_store.go:874]:
```suggestion
	size := 65534 / 2 // 65535 is the max number of parameters in a query, 2 is the number of parameters in each row. We round it down to get an integer.
```
I've checked the `SplitIntoChunks` code to see how it works with values like `32767.5`, but based on the code, I'm not sure, plus I think we could save a lot of wondering by just saying `65534/2` instead.

File: enterprise/internal/database/perms_store.go [hunk 2]
```diff
@@ -898,23 +910,56 @@ func (s *permsStore) SetRepoPermissionsUnrestricted(ctx context.Context, ids []i
...
 VALUES %s
-ON CONFLICT DO NOTHING`,
-		sqlf.Join(values, ","),
-	)
-	if !unrestricted {
-		q = sqlf.Sprintf(`DELETE FROM user_repo_permissions WHERE repo_id = ANY(%s)`, pq.Array(ids))
+ON CONFLICT DO NOTHING;
+`
+
+	size := 65535 / 4 // 65535 is the max number of parameters in a query, 4 is the number of parameters in each row
```
Comment (vdavid) [inline enterprise/internal/database/perms_store.go:949]: Similarly to above:
```suggestion
	size := 65532 / 4 // 65535 is the max number of parameters in a query, 4 is the number of parameters in each row. Rounding it down to get a whole number.
```

File: enterprise/internal/database/perms_store.go [hunk 3]
```diff
@@ -898,23 +910,56 @@ func (s *permsStore) SetRepoPermissionsUnrestricted(ctx context.Context, ids []i
...
 	}
 
+	if !unrestricted {
+		return txs.unsetRepoPermissionsUnrestricted(ctx, ids)
+	}
+	return txs.setRepoPermissionsUnrestricted(ctx, ids)
+}
+
+func (s *permsStore) unsetRepoPermissionsUnrestricted(ctx context.Context, ids []int32) error {
+	format := `DELETE FROM user_repo_permissions WHERE repo_id = ANY(%s) AND user_id IS NULL;`
```
Comment (vdavid) [inline enterprise/internal/database/perms_store.go:920]: That `AND user_id IS NULL` add was hard for me to spot. Pointing it out for other reviewers.

**Review by vdavid — APPROVED** [review]
LGTM. To be honest, I didn't go through the test.

File: enterprise/internal/database/perms_store.go (hunk 1 above)
Comment (kopancek) [reply enterprise/internal/database/perms_store.go:874]: It's go, so integer division leads to what would in javascript be `math.floor(65535/2)`.

File: enterprise/internal/database/perms_store.go (hunk 3 above)
Comment (kopancek) [reply enterprise/internal/database/perms_store.go:920]: yes, it is the most important bit here probably

**Pull request merged**